test/blackbox/common/mesh_event_handler.c

```diff
@@ -271,3 +271,22 @@ void mesh_event_destroy(void) {
 	pthread_cancel(event_receive_thread);
 }
 
+bool change_state(node_status_t *status, mesh_event_t currentEv) {
+
+	if(status->status) {
+		return false;
+	}
+
+	if(currentEv == status->expected_events[status->current_index]) {
+		status->current_index = status->current_index + 1;
+		status->currentEv = currentEv;
```

Comment:
This is the only place where `status->currentEv` is set, nothing is reading it anymore as far as I can see. So this can be removed.

Reply:
In every test case, there is a function called `signal_node_start()`. This function is going to read this currentEv variable in the structure.

```
static void signal_node_start(node_status_t *node_status, int start, int end) {
	int i;

	for(i = start; i <= end; i++) {
		if(NODE_JOINED != node_status[i].currentEv) {
			return;
		}
	}


	for(i = start; i <= end; i++) {
		node_step_in_container(node_ids[i], "SIGIO");
	}

	return;
}
```

This function waits for all the nodes to join the network and then sends signals to all the nodes to start channel connections and data exchange.

Reply:
If the function is exactly the same in each testcase, it should only be defined once. So I would move `signal_node_start()` into `mesh_event_handler.c`. Then, accounting for the fact that `current_index` points to the next expected event, you can replace:
```
if(NODE_JOINED != node_status[i].currentEv) {
```
With:
```
int index = node_status[i].current_index;
if(index < 1 || NODE_JOINED != node_status[i].expected_events[index - 1]) {
```